### PR TITLE
Necessary bits for packaging purposes

### PR DESCRIPTION
**Pull request by Spredzy**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+DEBUILD = $(DEBUILD_BIN) $(DEBUILD_OPTS)
+DEB_PPA ?= mini_dinstall
+DEB_ARCH ?= amd64
+DEB_NVR = $(NAME)_$(VERSION)-$(RELEASE)~$(DEB_DIST)
+DEB_NVRA = $(DEB_NVR)_$(DEB_ARCH)
+DEB_NVRS = $(DEB_NVR)_source
+DEB_TAR_NAME=$(NAME)-$(VERSION)
+DEB_TAR_FILE=$(NAME)_$(VERSION).orig.tar.gz
+DEB_DATE := $(shell LC_TIME=C date +"%a, %d %b %Y %T %z")
```
Comment (shanemcd) [inline Makefile]:
```suggestion
```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+.PHONY: clean dist sdist dev shell image devimage rpm srpm docs deb debian deb-src
```
Comment (shanemcd) [inline Makefile]:
```suggestion
.PHONY: clean dist sdist dev shell rpm srpm
```

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+clean:
+	rm -rf dist
+	rm -rf build
+	rm -rf ansible-builder.egg-info
+	rm -rf rpm-build
+	rm -rf deb-build
```
Comment (shanemcd) [inline Makefile]:
```suggestion
```

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+	mkdir -p $(dir $@)
+	@if [ "$(OFFICIAL)" != "yes" ] ; then \
+	  tar -C deb-build/ -xvf dist/$(NAME)-$(VERSION).tar.gz ; \
+	  cd deb-build && tar czf $(DEB_TAR_FILE) $(NAME)-$(VERSION) ; \
+	else \
+	  cp -a dist/$(NAME)-$(VERSION).tar.gz deb-build/$(DEB_TAR_FILE) ; \
+	fi
+	cd deb-build && tar -xf $(DEB_TAR_FILE)
+	cp -a packaging/debian deb-build/$(NAME)-$(VERSION)/
+	sed -ie "s|%VERSION%|$(VERSION)|g;s|%RELEASE%|$(RELEASE)|;s|%DEB_DIST%|$(DEB_DIST)|g;s|%DATE%|$(DEB_DATE)|g" $@/debian/changelog
```
Comment (shanemcd) [inline Makefile]:
```suggestion
```

**Review by shanemcd — COMMENTED** [review]
Let's remove the references to Debian packaging, I dont want to do this until we need to.

**Comment by shanemcd** [comment]
When I run this locally, it errors with the following:


```
$ make rpm                                                                                                                                                                                                                                                                                                  2 ↵
MOCK_CONFIG=epel-7-x86_64 docker-compose -f packaging/rpm/docker-compose.yml build
Building rpm-builder
Step 1/5 : FROM centos:7
 ---> b5b4d78bc90c
Step 2/5 : RUN yum install -y epel-release
 ---> Using cache
 ---> 6b26777876e0
Step 3/5 : RUN yum install -y make mock python3 which git gcc python3-devel
 ---> Using cache
 ---> 0a381069be4b
Step 4/5 : RUN sed -i "s/.el7.centos/.el7/g" /etc/rpm/macros.dist
 ---> Using cache
 ---> 2721f44540f9
Step 5/5 : RUN pip3 install -IU poetry ansible dephell
 ---> Using cache
 ---> 7d3be58a0e0f
Successfully built 7d3be58a0e0f
Successfully tagged builder-rpm-builder:epel-7-x86_64
MOCK_CONFIG=epel-7-x86_64 docker-compose -f packaging/rpm/docker-compose.yml \
  run --rm -e RELEASE=0.git2020052614.c402967 rpm-builder "make mock-rpm"
dephell deps convert --from=pyproject.toml --to=setup.py
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/dephell/imports.py", line 19, in _module
    return import_module(name=self._name)
  File "/usr/lib64/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'appdirs'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/dephell", line 7, in <module>
    from dephell.cli import entrypoint
  File "/usr/local/lib/python3.6/site-packages/dephell/cli.py", line 11, in <module>
    from .commands import COMMANDS
  File "/usr/local/lib/python3.6/site-packages/dephell/commands/__init__.py", line 2, in <module>
    from .discover import COMMANDS
  File "/usr/local/lib/python3.6/site-packages/dephell/commands/discover.py", line 86, in <module>
    module = import_module('.' + module_name, package=package)
  File "/usr/lib64/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/usr/local/lib/python3.6/site-packages/dephell/commands/deps_add.py", line 7, in <module>
    from ..actions import get_resolver
  File "/usr/local/lib/python3.6/site-packages/dephell/actions/__init__.py", line 7, in <module>
    from ._converting import attach_deps
  File "/usr/local/lib/python3.6/site-packages/dephell/actions/_converting.py", line 6, in <module>
    from ..config import Config
  File "/usr/local/lib/python3.6/site-packages/dephell/config/__init__.py", line 3, in <module>
    from .manager import Config
  File "/usr/local/lib/python3.6/site-packages/dephell/config/manager.py", line 20, in <module>
    from .defaults import DEFAULT
  File "/usr/local/lib/python3.6/site-packages/dephell/config/defaults.py", line 35, in <module>
    venv=str(get_data_dir() / 'venvs' / '{project}-{digest}' / '{env}'),
  File "/usr/local/lib/python3.6/site-packages/dephell/config/app_dirs.py", line 31, in get_data_dir
    return Path(appdirs.user_data_dir(app))
  File "/usr/local/lib/python3.6/site-packages/dephell/imports.py", line 43, in __getattr__
    return getattr(self._module, name)
  File "/usr/local/lib/python3.6/site-packages/dephell/imports.py", line 21, in _module
    self._install()
  File "/usr/local/lib/python3.6/site-packages/dephell/imports.py", line 25, in _install
    from .package_manager import PackageManager
  File "/usr/local/lib/python3.6/site-packages/dephell/package_manager.py", line 14, in <module>
    from .converters import PIPConverter
  File "/usr/local/lib/python3.6/site-packages/dephell/converters/__init__.py", line 5, in <module>
    from .base import BaseConverter
  File "/usr/local/lib/python3.6/site-packages/dephell/converters/base.py", line 11, in <module>
    from ..controllers import Graph, Mutator, Resolver
  File "/usr/local/lib/python3.6/site-packages/dephell/controllers/__init__.py", line 3, in <module>
    from ._dependency import DependencyMaker
  File "/usr/local/lib/python3.6/site-packages/dephell/controllers/_dependency.py", line 14, in <module>
    from ..models.constraint import Constraint
  File "/usr/local/lib/python3.6/site-packages/dephell/models/__init__.py", line 5, in <module>
    from .dependency import Dependency
  File "/usr/local/lib/python3.6/site-packages/dephell/models/dependency.py", line 13, in <module>
    from .group import Group
  File "/usr/local/lib/python3.6/site-packages/dephell/models/group.py", line 7, in <module>
    from ..config import config
ImportError: cannot import name 'config'
make: *** [dist/ansible-builder-0.1.0.tar.gz] Error 1
make: *** [Makefile:68: rpm] Error 2
```

**Comment by Spredzy** [comment]
recheck

**Comment by Spredzy** [comment]
@shanemcd issue has been fixed with latest patchset. Please re-try and thanks for bringing it up.

**Comment by shanemcd** [comment]
I just tried again and I'm seeing the same error. 😕

**Review by shanemcd — APPROVED** [review]

**Pull request merged**